hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/util/subject/HiddenGetSubject.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.util.subject;
+
+import javax.security.auth.Subject;
+
+public interface HiddenGetSubject {
```

Comment:
there were three methods in Subject marked as `@Deprecated(since="17", forRemoval=true)`, can we make the interface name more generic?

Reply:
so renamed in latest commit:
```
ClassicGetSubject.java => ClassicSubjectAdapter.java
HiddenGetSubject.java => HiddenSubjectAdapter.java
GetSubjectJava18AndAbove.java => SubjectAdapterJava18AndAbove.java
```

I did initially write the "doAs" support, but backed out of it when I encountered public exposure in UserGroupInformation.  I do not understand the fate and timeline for removal, if removal is the ultimate fate, of things like PrivilegeExceptionAction.